src/app/data_harvester.rs

```diff
@@ -288,17 +293,96 @@ impl DataCollector {
+            #[cfg(feature = "nvidia")]
+            if let Some(data) = nvidia::get_nvidia_vecs(
+                &self.temperature_type,
+                &self.filters.temp_filter,
+                use_temp,
+                use_mem,
+                use_proc,
+                use_cpu,
+                use_battery,
+            ) {
+                if use_temp {
+                    if let Some(mut temp) = data.temperature {
+                        if let Some(ref mut sensors) = self.data.temperature_sensors {
+                            sensors.append(&mut temp);
+                        } else {
+                            self.data.temperature_sensors = Some(temp);
+                        }
+                    }
+                }
+                if use_mem {
+                    if let Some(mem) = data.memory {
+                        self.data.gpu = Some(mem);
+                    }
+                }
+                if use_proc {
+                    if let Some(proc) = data.procs {
+                        self.gpu_pids = Some(proc);
+                    }
+                }
+                if use_cpu {
+                    if let Some(mut cpu) = data.usage {
+                        if let Some(ref mut cpus) = self.data.cpu {
+                            cpus.append(&mut cpu);
+                        } else {
+                            self.data.cpu = Some(cpu);
+                        }
+                    }
+                }
+                #[cfg(all(feature = "battery", target_os = "linux"))]
+                {
+                    use crate::data_harvester::batteries::BatteryHarvest;
+                    use starship_battery::State;
+                    if use_battery {
+                        if let Some(power) = data.battery {
+                            let mut powers: Vec<BatteryHarvest> = power
+                                .into_iter()
+                                .map(|pwr| {
+                                    BatteryHarvest {
+                                        charge_percent: 100.0,
+                                        secs_until_full: None,
+                                        secs_until_empty: None,
+                                        power_consumption_rate_watts: (pwr.1 / 1000) as f64, // convert milliwatts to watts
+                                        health_percent: 100.0,
+                                        state: State::Unknown,
+                                        name: pwr.0,
+                                    }
+                                })
+                                .collect();
+                            if let Some(ref mut batts) = self.data.list_of_batteries {
+                                batts.append(&mut powers);
+                            } else {
+                                self.data.list_of_batteries = Some(powers);
+                            }
+                        }
+                    }
+                }
```

Comment:
Hm, not sure how I feel about this part - power consumption is an interesting stat for sure, but I feel like sticking it as a battery metric feels a bit like a wrong move? Might be better as a new widget type or something.

Reply:
I am fine with removing it. It can be added in a different widget later...